jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/UrlResourceFactory.java

```diff
@@ -0,0 +1,240 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util.resource;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.file.Path;
+import java.time.Instant;
+
+import org.eclipse.jetty.util.FileID;
+
+/**
+ * {@link ResourceFactory} for {@link java.net.URL} based resources.
+ */
+class UrlResourceFactory implements ResourceFactory
+{
+    private final String supportedProtocol;
+    private int connectTimeout;
+    private boolean useCaches;
+
+    protected UrlResourceFactory(String protocol)
+    {
+        this.supportedProtocol = protocol;
+        this.connectTimeout = Integer.parseInt(System.getProperty(this.getClass().getName() + ".connectTimeout", "1000"));
+        this.useCaches = Boolean.parseBoolean(System.getProperty(this.getClass().getName() + ".useCaches", "true"));
```

Comment:
Let's not continue using System Properties for configuration.  There are setters available.